iox_catalog/src/postgres.rs

```diff
@@ -1287,6 +1287,18 @@ WHERE table_id = $1;
         .map_err(|e| Error::SqlxError { source: e })
     }
 
+    async fn list_ids(&mut self) -> Result<Vec<PartitionId>> {
+        sqlx::query_as(
+            r#"
+            SELECT p.id as partition_id
+            FROM partition p
+            "#,
```

Comment:
```suggestion
           WHERE to_delete IS NULL,
            "#,
```

I think there is a to_delete flag on partitions too:

```sql
id  shard_id  table_id  partition_key  sort_key  persisted_sequence_number  to_delete  new_file_at
--  --------  --------  -------------  --------  -------------------------  ---------  -----------
1   1         5         2023-02-06     []
2   1         4         2023-02-06     []
3   1         1         2023-02-06     []
4   1         2         2023-02-06     []
5   1         3         2023-02-06     []
6   1         6         2023-02-06     []
```

Reply:
WTH, there is not even a way to set this marker :shrug: 

Let me drive-by clean this mess then...

Reply:
Yeah wait, I'm not fixing this. Nothing in our stack looks at this flag and I don't have the time now to clean it. It's not supported, full stop.

Reply:
#6905.